execute/plugin.go

```diff
@@ -151,25 +144,30 @@ func getPendingExecutedReports(
 			continue
 		}
 
+		sort.Slice(reports, func(i, j int) bool {
+			return reports[i].SequenceNumberRange.Start() < reports[j].SequenceNumberRange.Start()
+		})
+
 		ranges, err := computeRanges(reports)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("compute report ranges: %w", err)
 		}
 
-		var executedMessages []cciptypes.SeqNumRange
+		executedMessageSet := mapset.NewSet[cciptypes.SeqNum]()
 		for _, seqRange := range ranges {
-			executedMessagesForRange, err2 := ccipReader.ExecutedMessageRanges(ctx, selector, dest, seqRange)
+			executedMessagesForRange, err2 := ccipReader.ExecutedMessages(ctx, selector, dest, seqRange)
```

Comment:
Just another comment on this re: performance - DB-query wise, you'd probably just wanna do one query to cover all the different contiguous sub-ranges. This right now does a query on every single sub-range separately, which makes DB I/O quite heavy for this function/loop and probably much slower than it should be.

cc: @0xAustinWang @mateusz-sekara